be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/package.scala

```diff
@@ -5,4 +5,8 @@ import ch.epfl.pop.model.network.JsonRpcMessage
 package object graph {
   type JsonString = String
   type GraphMessage = Either[JsonRpcMessage, PipelineError] // FIXME convention states that Right(_) is the success value
+
+  def bindToPipe(rpcMessage: JsonRpcMessage, success: Boolean, error: PipelineError) = {
+    if (success) Left(rpcMessage) else Right(error)
+  }
```

Comment:
If this function is accepted, it would probably be good to wait for the other validators refactoring happening with respect to issue [#1402 ](https://github.com/dedis/popstellar/issues/1402) to be merged to master so as to their validators use this function too before completing the pull request.

Reply:
I came up with such a function at the beginning of the refactoring process but I gave up the idea as it kinda defeat the purpose of replacing the if-else pattern. My opinion is that the point of using these `checkSomething()` function is to have abstraction and having this `bindToPipe()` kinda short circuit that.